haddock-api/src/Haddock/Convert.hs

```diff
@@ -311,7 +311,7 @@ synifyDataCon use_gadt_syntax dc =
                          , con_doc =  Nothing }
 
 synifyName :: NamedThing n => n -> Located Name
-synifyName = noLoc . getName
+synifyName n = L (srcLocSpan (getSrcLoc n)) (getName n)
```

Comment:
@niteria I understand this change. It is a great one! We were previously assuming `noLoc` where we now use the location of a name, great 100% more information!